x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyoutBody.tsx

```diff
@@ -195,28 +228,26 @@ export function AddSettingFlyoutBody({
           isInvalid={
             !(
               (Boolean(selectedConfig) &&
-                (isNaN(sampleRate) || isSampleRateValid)) ||
+                (isNaN(sampleRateFloat) || isSampleRateValid)) ||
```

Comment:
Just wondering, can we not share the Joi schema that we use in the API?

Reply:
We don't currently use Joi on the client side. Joi isn't an isomorphic package as it relies heavily on native node modules. We could try to leverage something like `joi-browser` if we wanted to share schemas.